tests/batch/indexer_Position_Coupon_test.py

```diff
@@ -437,3 +476,137 @@ def test_error_1(self, processor, shared_contract, session):
+                pytest.raises(ServiceUnavailable):
+            processor.initial_sync()
+        # Assertion
+        _position_list = session.query(IDXPosition).order_by(IDXPosition.created).all()
+        assert len(_position_list) == 0
+        assert processor.latest_block == block_number_bf
+
+        # Transfer
+        coupon_transfer_to_exchange(
+            self.issuer, {"address": self.trader["account_address"]}, token, 10000)
+
+        block_number_bf = processor.latest_block
+        # Expect that sync_new_logs() raises ServiceUnavailable.
+        with mock.patch("web3.eth.Eth.block_number", side_effect=ServiceUnavailable()), \
+                pytest.raises(ServiceUnavailable):
+            processor.sync_new_logs()
+
+        # Assertion
+        _position_list = session.query(IDXPosition).order_by(IDXPosition.created).all()
+        assert len(_position_list) == 0
+        # Latest_block is NOT incremented in "sync_new_logs" process.
+        assert processor.latest_block == block_number_bf
+
+    # <Error_2_2>: SQLAlchemyError occurs in "initial_sync" / "sync_new_logs".
+    def test_error_2_2(self, processor, shared_contract, session):
+        # Issue Token
+        token_list_contract = shared_contract["TokenList"]
+        token = self.issue_token_coupon(
+            self.issuer, config.ZERO_ADDRESS, token_list_contract)
+        self.listing_token(token["address"], session)
+
+        # Transfer
+        coupon_transfer_to_exchange(
+            self.issuer, {"address": self.trader["account_address"]}, token, 10000)
+
+        block_number_bf = processor.latest_block
+        # Expect that initial_sync() raises SQLAlchemyError.
+        with mock.patch.object(Session, "commit", side_effect=SQLAlchemyError()), \
+                pytest.raises(SQLAlchemyError):
+            processor.initial_sync()
+
+        # Assertion
+        _position_list = session.query(IDXPosition).order_by(IDXPosition.created).all()
+        assert len(_position_list) == 0
+        assert processor.latest_block == block_number_bf
+
+        # Transfer
+        coupon_transfer_to_exchange(
+            self.issuer, {"address": self.trader["account_address"]}, token, 10000)
+
+        block_number_bf = processor.latest_block
+        # Expect that sync_new_logs() raises SQLAlchemyError.
+        with mock.patch.object(Session, "commit", side_effect=SQLAlchemyError()), \
+                pytest.raises(SQLAlchemyError):
+            processor.sync_new_logs()
+
+        # Assertion
+        _position_list = session.query(IDXPosition).order_by(IDXPosition.created).all()
+        assert len(_position_list) == 0
+        # Latest_block is NOT incremented in "sync_new_logs" process.
+        assert processor.latest_block == block_number_bf
+
+    # <Error_3>: ServiceUnavailable occurs and is handled in mainloop.
+    def test_error_3(self, main_func, shared_contract, session, caplog):
```

Comment:
When some exceptions occur in “sync_new_logs”(test_error_3), batch will output logs like below.

```
[2022-05-17 11:28:57 +0900] [INDEXER-POSITION-BOND] [90226] [INFO] Service started successfully
[2022-05-17 11:28:58 +0900] [INDEXER-POSITION-BOND] [90226] [ERROR] An exception occurred during event synchronization
Traceback (most recent call last):
  ~StackTrace
app.errors.ServiceUnavailable
[2022-05-17 11:28:58 +0900] [INDEXER-POSITION-BOND] [90226] [WARNING] An external service was unavailable
```